src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -293,10 +293,19 @@
           (app (assoc req :body-string payload))
           (http/error-response (str "Missing required parameter 'payload'")))
         "application/json"
-        (let [body-string (request/body-string req)]
+        (let [body-string (request/body-string req)
+              reconstructed-body (assoc params "payload" body-string)
+              param-post? (boolean (and (get params "certname")
+                                        (get params "version")
+                                        (get params "command")))]
```

Comment:
It feels like if they specify any one of these, that we should error if they don't have all of them.